recon_surf/N4_bias_correct.py

```diff
@@ -0,0 +1,304 @@
+#!/usr/bin/env python3
+
+
+# Copyright 2019 Image Analysis Lab, German Center for Neurodegenerative Diseases (DZNE), Bonn
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# IMPORTS
+import optparse
+import numpy as np
+import sys
+import SimpleITK as sitk
+import nibabel as nib
+import image_io as iio
+
+
+HELPTEXT = """
+
+Script to call SITK N4 Bias Correction
+
+USAGE:
+N4_bias_correct  --invol <img.nii> --outvol <corrected.nii> OPTIONS
+
+
+Dependencies:
+    Python 3.6
+
+    SimpleITK https://simpleitk.org/ (v2.1.1)
+
+
+N4 Bias Correction:
+
+The input image will be N4 bias corrected. Optimally you would pass a brainmask via
+--mask. If no mask is given, all 0 values will be masked to speed-up correction and
+avoid influence of flat zero regions on the bias field. 
+
+
+WM Normalization and UCHAR:
+
+Unless --skipwm is passed the output will be converted to UCHAR.
+For this a ball of radius 50 voxels at the center of the image is used to find the
+1st and 90th percentile of intensities in the brain. This region is then mapped to
+3 .. 110, so that (WM) will be around 110.
+
+There are different options how the center is found:
+- If a mask (e.g. brainmask) is passed, the centroid of the mask will be used and 
+  the ball will be masked to crop non-brain tissue.
+- If no mask is passed, the OTSU method is used to find a brain mask and centroid. 
+  Note, that the ball will not be cropped and may contain non-brain tissue.
+- If a talairach.xfm transform is passed, we use the origin of the Talairach space
+  as center for the ball. If additionally a mask was passed, the ball will be 
+  masked to crop non-brain tissue. 
+
+
+
+Original Author: Martin Reuter
+Date: Mar-18-2022
+"""
+
+h_invol    = 'path to input.nii.gz'
+h_outvol   = 'path to corrected.nii.gz'
+h_mask     = 'optional: path to mask.nii.gz'
+h_shrink   = '<int> shrink factor, default: 4'
+h_levels   = '<int> number of fitting levels, default: 4'
+h_numiter  = '<int> number of iterations per level, default: 4'
```

Comment:
Here, it says default: 4, but in the parser option below (line 92) the default is set to 50 (probably copy-paste error from h_levels?)